doc/feature/ReqRep.md

```diff
@@ -0,0 +1,112 @@
+Request-Reply {#reqrep}
+============
+
+Certain operations are by necessity synchronous, RPC-like exchanges. This
+specification describes this client-server API based on zmq req-rep (dealer/router) sockets.
+
+## Requirements
+
+* Asynchronous remote procedure call from a Client to a Server
+
+## API
+
+    using ReplyFunc = std::function<void(const void*, size_t)>;
+
+    class Client : public Receiver
+    {
+        // ctors same as Subscriber
+        void get(const Serializable& request, ReplyFunc& func);
+    };
+
+    using HandleFunc = std::function<servus::Serializable::Data(const void*, size_t)>;
+
+    class Server : public Receiver
+    {
+        // ctors as Publisher
+        void handle(const uint128_t& request, HandleFunc& func)
+    };
+
+
+## Implementation
+
+* Maps closely to ZeroMQ
+  * Client uses a dealer socket, receive() will call ReplyFunc
+  * Server::receive() uses a rep socket
+    * Single-threaded, synchronous for now
+    * Can be extended to async model later
+    * does recv()-send()
+* Client::get is fully asynchronous
+* Client::get will provide nullptr, 0 if server has no handle func or handle func returns no data
+
+## Examples
+
+    class livre::RemoteDataSource
+    {
+        RemoteDataSource(const DataSourcePluginData& initData)
+        {
+            if( !_client.get( initData.uri, initData.info ))
+                throw std::runtime_error( "Open failed");
+        }
+
+        MemoryUnitPtr getData(const LODNode& node)
+        {
+            MemoryUnitPtr block;
+            _client.get({node, _uri},
+                        [&]( const void* data, const size_t size ){
+                            block = new AllocMemoryUnit(size);
+                            memcpy data to block
+                        });
+
+            _client.receive( TIMEOUT );
```

Comment:
I didn't even think about it because that's an implementation detail. I'd argue that mixing request/reply (which are synchronous) with pub/sub receivers or even receivers from different servers doesn't look like a good idea. I don't know about the implementation details from zmq, but isn't the call to receive the reply blocking? Can you still poll multiple request sockets for the reply?